src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -122,6 +125,9 @@ function ShallowWaterModel(;
                                        "Use `topology = ($(topology(grid, 1)), $(topology(grid, 2)), Flat)` " *
                                        "when constructing `grid`."
 
+    @assert ((typeof(grid) <: RectilinearGrid) && formulation == ConservativeFormulation()) "`ConservativeFormulation()` requires a rectilinear `grid`. \n" *
+                                                                                            "Use `VectorInvariantFormulation()` or change your grid to a rectilinear one."
```

Comment:
Note that `@assert` can be disabled for some optimization levels so this may not be what you want:

https://docs.julialang.org/en/v1/base/base/#Base.@assert

Perhaps you are looking for `ArgumentError`?

Reply:
how would that be called?

`if condition; ArgumentError("message"); end`

?

Reply:
no, better with short-circuit

Reply:
```Julia
(!(typeof(grid) <: RectilinearGrid) && formulation == ConservativeFormulation()) && ArgumentError("...")
```

Reply:
It's

```julia
throw(ArgumentError(msg))
```